Assets/Scripts/Models/Room.cs

```diff
@@ -19,18 +19,49 @@
 public class Room : IXmlSerializable
 {
     // Dictionary with the amount of gas in room stored in preasure(in atm) multiplyed by number of tiles.
-    private Dictionary<string, float> atmosphericGasses; 
+    private Dictionary<string, float> atmosphericGasses;
     private Dictionary<string, string> deltaGas;
 
     private List<Tile> tiles;
 
+    // TEST code, should be removed before PR is merged!
+    static Room()
+    {
+        OnRoomCreated += (r) =>
+            {
+                Debug.ULogErrorChannel("Room", "Room Made: {0}", r.ID);
+            };
+        OnRoomSplit += (r, r2) =>
+            {
+                Debug.ULogErrorChannel("Room", "Room Split: {0}, {1}", r.ID, r2.ID);
+            };
+        OnRoomJoined += (r, r2) =>
+            {
+                // FIXME: By the time the rooms are joined the information about their respective IDs is gone.
+                // Not sure what to do about it, do we need tht information for this event?
+                Debug.ULogErrorChannel("Room", "Room Joined: {0}, {1}", r.ToString(), r2.ID);
+            };
+        OnRoomDestroyed += (r) =>
+            {
+                Debug.ULogErrorChannel("Room", "Room Broken: {0}", r.ID);
+            };
+    }
+
     public Room()
     {
         tiles = new List<Tile>();
         atmosphericGasses = new Dictionary<string, float>();
         deltaGas = new Dictionary<string, string>();
     }
 
+    public static event Action<Room> OnRoomCreated;
```

Comment:
I have it as `static` because otherwise you would have to set each of the events on a per `Room` basis, I don't think anyone actually wants to do that, therefore they are static, the other option is having something like a `RoomManager` of sorts that handles all of that, in which case these could be non-`static` but part of the instance of that manager.